### DISCUSSION TITLE
[Chapter 8] Paper Replicating, position embedding

### DISCUSSION THREAD

**Discussion by ivantronics** [mrdbourke/pytorch-deep-learning#411] [Q&A]
Hello, @mrdbourke 
Thank you very much for your amazing course, I am learning a lot and I love your enthusiasm when you code, I try to get infected by it as much as possible!

So far everything has been pretty clear and understandable, but in Chapter 8 we add position embedding to our tensor with class token prepended and tensor for each patch. I am a little bit confused by what's going on and in the image from the paper itself.
So we have our patches. We want our network to know (or somehow understand) the position of each patch so we want to add some data about that into our tensor like we did with class token.
I was under the impression that we will have some kind of additional dimension that will store this data (again, like class token). But we just add some kind of arbitrary (ones in video, but it will be random) value to each of the embedding tensor's parameter? How does that work? How does our network know about the position of each patch from that operation? Wouldn't it just modify our existing patches into something else, especially if every value would be random?
Image from paper shows purple numbers assigned to each tensor with the patch embeddings.
![изображение](https://user-images.githubusercontent.com/7630655/233136508-43fd415f-ecb7-4073-8876-97ba9028b7a5.png)

So when we prepare our image, how do we obtain these values for positions exactly? From reading the paper I thought it will be just some kind of positional data like coordinates or something. If it's random, then how does our network learn from it? I mean, I get that we add it as a parameter with gradient tracking, so in theory our network can train this parameter. Is this what's partially happening in transformer encoder next? How does our network understand that every value had some kind of number added to it beforehand? 
Shouldn't we tell it exaclty where every patch belongs?
Or have entirely separate dimension to store that data instead of just adding it broadly over our patch embeddings?

I am sorry if this is a dumb question. I got confused how that works.
Thank you again for the wonderful course, I am super inspired by it
